target-i386/trace_helper.c

```diff
@@ -184,3 +169,72 @@ void HELPER(trace_st)(CPUArchState *env, target_ulong val, target_ulong addr)
 
     qemu_trace_add_operand(oi, 0x2);
 }
+
+
+OperandInfo * store_flag(short val, const char *reg_name)
+{
+    RegOperand * ro = g_new(RegOperand, 1);
+    reg_operand__init(ro);
+        
+    ro->name = g_strdup(reg_name);
+    strcpy(ro->name, reg_name);
+
+    OperandInfoSpecific *ois = g_new(OperandInfoSpecific, 1);
+    operand_info_specific__init(ois);
+    ois->reg_operand = ro;
+
+    OperandUsage *ou = g_new(OperandUsage, 1);
+    operand_usage__init(ou);
+    ou->written = 1;
+    OperandInfo *oi = g_new(OperandInfo, 1);
+    operand_info__init(oi);
+    oi->bit_length = 1;
+    oi->operand_info_specific = ois;
+    oi->operand_usage = ou;
+    oi->value.len = 1;
+    oi->value.data = g_malloc(oi->value.len);
+    memcpy(oi->value.data, &val, 1);
+    return oi;
+}
+
+bool is_flag_changed(x86_flags_t old_flags, x86_flags_t flags, x86_flags_t mask) {
+    return (old_flags & mask) != (flags & mask);
+}
+
+short flag_value(x86_flags_t flags, x86_flags_t mask) {
+    if ((flags & mask) == mask) return 1;
+    else return 0;
+}
+
+void store_if_changed(x86_flags_t old_flags, x86_flags_t flags, x86_flags_t mask, const char *reg_name) {
+    if (is_flag_changed(old_flags, flags, mask)) {
+        OperandInfo *oi = store_flag(flag_value(flags, mask), reg_name);
+        qemu_trace_add_operand(oi, 0x2);
+    }
+}
+
+void set_x86_flags(x86_flags_t flags) {
+    memcpy(state(), &flags, sizeof(x86_flags_t));
+}
+
+x86_flags_t x86_flags(void) {
+    x86_flags_t flags = 0;
+    if (state() == NULL) { 
+        void *state = g_malloc(sizeof(x86_flags_t));
+        set_state(state);
+    } else {
+        memcpy(&flags, state(), sizeof(x86_flags_t));
+    }
+    return flags;
+}
+
+void HELPER(trace_store_eflags)(CPUArchState *env)
+{
+    x86_flags_t old_flags = x86_flags();
+    x86_flags_t flags = cpu_compute_eflags(env);
+    set_x86_flags(flags);
+    store_if_changed(old_flags, flags, CC_C, "CF");
```

Comment:
you should follow the established tradition for handling registers. If you have a desire to introduce you own, then you should fix the old code, so that it follows your new convention. So, in the established convention, we add `enum` for flags, and write a mapping from each flag to the text representation.